tests/test_pipelines_token_classification.py

```diff
@@ -61,6 +64,17 @@ def _test_pipeline(self, token_classifier: Pipeline):
             for key in output_keys:
                 self.assertIn(key, result)
 
+    @require_torch
+    def test_model_kwargs_passed_to_model_load(self):
+        # Tests the model_kwargs parameter is passed properly when loading a model with pipeline()
+        # Currently uses the cache_dir parameter of from_pretrained()
+        # There might be a better parameter, but this one is fairly easy to test.
```

Comment:
Indeed, this one is fairly easy to test - but it implies redownloading the model, which adds quite a bit of overhead.

Another test that could be done is to test that a configuration attribute is correctly propagated through the model. For example the following:

```py
ner_pipeline = pipeline(task="ner", model=self.small_models[0], model_kwargs={"output_attentions": True})
self.assertTrue(ner_pipeline.model.config.output_attentions)
```

This way the model doesn't get redownloaded, instead using the cached version; but we're still ensuring that the model kwargs are correctly consumed.

Reply:
I not a fan of needing to redownload the model, but looking at [the variables read from `kwargs` in `PreTrainedModel.from_pretrained()`](https://github.com/huggingface/transformers/blob/f0dde6012784c79155f7c1bd721864ea8ba40821/src/transformers/modeling_utils.py#L1118-L1134), most of them require network calls for complicated tests.

I wasn't aware of the other `model_kwargs` field you included, but that results in a much easier test! Thanks for the help!

Reply:
You're right, most of them are related to network calls! When passing other arguments, they're used for the model configuration, such as here.

Some values, like `output_attentions` and `output_hidden_states`, or `return_dict`, are common to all models so testing that here is robust :)

Thanks for updating the code!